### PR TITLE
[15611] Lock orientation on sign in screen

### LINKED ISSUE #15611: Screen orientation change is enabled on Android (list of cases)
@ibrkhalil I have updated the issue and added couple more cases when screen orientation change gets enabled on Android. 

Android only. Tested on Samsung Galaxy A52, Android 12

**Case 1:** Orientation change is enabled on Sign in by syncing onboarding 

**Steps:** 

1. Open Sign in by syncing onboarding screen
2. Try to scan invalid QR code 
3. Try to change screen mode to landscape at the moment when error "Oops! This is not a sync QR code" appears 
4. See if it is possible to switch to landscape mode

**Actual result:** can switch to landscape mode. 

https://user-images.githubusercontent.com/97245802/230921887-91bc2881-e195-4617-adb8-6aed7dfe3d25.mp4

**Case 2:** Orientation change is enabled when Request to join community/New chat bottom sheets are opened

https://user-images.githubusercontent.com/97245802/235913260-3fb26ade-9661-4f8e-842d-b4812b81d198.mp4

https://user-images.githubusercontent.com/97245802/235913267-9016cd83-4317-44c3-8b9f-eb3334bcf1cf.mp4

**Case 3:** Orientation change is enabled when notification toast appears

https://user-images.githubusercontent.com/97245802/235913497-6fddd5d2-f2a8-4fb3-800a-a5645963d8a2.mp4

### Additional Information

- Status version: nightly 
- Operating System: Android

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #15611 

### Summary
Disabled orientation change on sign in page.

status: ready

### CODE REVIEW

**Comment by smohamedjavid** [comment]
Hi @ibrkhalil - Thanks for looking into this bug.

I believe we can lock the orientation from the screens (`src/status_im2/navigation/screens.cljs`) for `:sign-in` by passing `:options {:layout {:orientation ["portrait"]}`.

**Comment by ibrkhalil** [comment]
> navigation/screens

Thanks, Mohamed.
Didn't know we had such a configuration available, Done.

**Review by smohamedjavid — APPROVED** [review]

**Review by ilmotta — APPROVED** [review]

**Comment by OmarBasem** [comment]
I think all screens should be locked to `portrait` by default. In `status-im2.navigation.options` it is applied in the default options.

**Comment by pavloburykh** [comment]
Hey @ibrkhalil! Thanx for the PR. 

Still reproducing the bug. Orientation change is being unlocked when error "Oops! This is not a sync QR code" appears


https://user-images.githubusercontent.com/97245802/231377136-5858e4ff-c52f-4e0b-96ca-b45d8ef7b4b3.mp4

File: src/status_im2/navigation/screens.cljs [hunk 1]
```diff
@@ -116,7 +116,8 @@
      :component enable-notifications/enable-notifications}
 
     {:name      :sign-in
-     :component sign-in/view}
+     :component sign-in/view
+     :options   {:layout {:orientation ["portrait"]}}}
```
Comment (flexsurfer) [inline src/status_im2/navigation/screens.cljs]: portrait is the default option for all screens  so it shouldn't be set here, seem like somehow this setting was overwritten for this screen, by some overlay or ...

**Comment by smohamedjavid** [comment]
> I think all screens should be locked to `portrait` by default. In `status-im2.navigation.options` it is applied in the default options.

I agree with @OmarBasem.

I think the bug was caused by `default-options` as we pass a keyword to `:orientation` instead of passing a vector with a list of supported orientations. 

```clojure
(defn default-options
  []
  {:layout {:orientation :portrait}
   :topBar {:visible false}})

```
https://wix.github.io/react-native-navigation/docs/style-orientation

File: src/status_im2/navigation/screens.cljs (hunk 1 above)
Comment (ibrkhalil) [reply src/status_im2/navigation/screens.cljs]: Will investigate more

**Comment by ibrkhalil** [comment]
> > I think all screens should be locked to `portrait` by default. In `status-im2.navigation.options` it is applied in the default options.
> 
> I agree with @OmarBasem.
> 
> I think the bug was caused by `default-options` as we pass a keyword to `:orientation` instead of passing a vector with a list of supported orientations.
> 
> ```clojure
> (defn default-options
>   []
>   {:layout {:orientation :portrait}
>    :topBar {:visible false}})
> ```
> 
> https://wix.github.io/react-native-navigation/docs/style-orientation

Will investigate more

**Comment by flexsurfer** [comment]
> I think the bug was caused by `default-options` as we pass a keyword to `:orientation` instead of passing a vector with a list of supported orientations.

oh good catch, i was wondering why it didn't work well :)

File: src/status_im2/navigation/options.cljs [hunk 2]
```diff
@@ -4,7 +4,7 @@
 
 (defn default-options
   []
-  {:layout {:orientation :portrait}
+  {:layout {:orientation ["portrait"]}
```
Comment (flexsurfer) [inline src/status_im2/navigation/options.cljs:7]: actually, we should keep only this one, and remove orientation prop in other places, and you can keep it as keyword `:portrait`

Comment (ibrkhalil) [reply src/status_im2/navigation/options.cljs:7]: Wasn't the issue that we're using a keyword instead a of vector of strings ?

Comment (flexsurfer) [reply src/status_im2/navigation/options.cljs:7]: no just keyword instead of vector

File: src/status_im2/navigation/options.cljs [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 (defn default-options
   []
-  {:layout {:orientation ["portrait"]}
+  {:layout {:orientation :portrait}
```
Comment (flexsurfer) [inline src/status_im2/navigation/options.cljs]: it should be a vector`{:layout {:orientation [:portrait]}`

Comment (flexsurfer) [reply src/status_im2/navigation/options.cljs]: yeah sorry it doesn't look good, lets keep first variant with the string, sorry for confusion

Comment (flexsurfer) [reply src/status_im2/navigation/options.cljs]: `{:layout {:orientation ["portrait"]}`

Comment (ibrkhalil) [reply src/status_im2/navigation/options.cljs]: No worries, Thanks for reviewing :)

**Review by flexsurfer — APPROVED** [review]

**Comment by pavloburykh** [comment]
@ibrkhalil hello! JFYI I have updated the issue https://github.com/status-im/status-mobile/issues/15611. Added couple of more cases when screen orientation change gets enabled on Android.

**Comment by flexsurfer** [comment]
hey @ibrkhalil any updates on this?

**Comment by ibrkhalil** [comment]
> hey @ibrkhalil any updates on this?

I commented [here](https://github.com/status-im/status-mobile/issues/15611#issuecomment-1547217076) that I am blocked on this task because I don't have access to a physical android device atm.

**Comment by flexsurfer** [comment]
> I commented [here](https://github.com/status-im/status-mobile/issues/15611#issuecomment-1547217076) that I am block on this task because I don't have access to a physical android device atm.

but this should be reproducible on emulator as well

**Comment by ibrkhalil** [comment]
> > I commented [here](https://github.com/status-im/status-mobile/issues/15611#issuecomment-1547217076) that I am block on this task because I don't have access to a physical android device atm.
> 
> but this should be reproducible on emulator as well

How can I scan QR code on emulator (to view the bug) ?

**Comment by flexsurfer** [comment]
> How can I scan QR code on emulator (to view the bug) ?

just insert the text of QR code

**Comment by ibrkhalil** [comment]
> > How can I scan QR code on emulator (to view the bug) ?
> 
> just insert the text of QR code

I need to use the camera to scan an invalid QR code to be able to tilt the device horizontally while scanning to see the issue.

**Comment by flexsurfer** [comment]
> I need to use the camera to scan an invalid QR code to be able to tilt the device horizontally while scanning to see the issue.

i looked at the video and i don't think you need to scan, you just need to open qr modal and "rotate" emulator, it seems like modal screens for some reason unlock rotation

**Comment by pavloburykh** [comment]
> i looked at the video and i don't think you need to scan, you just need to open qr modal and "rotate" emulator, it seems like modal screens for some reason unlock rotation

bug is reproduced in the moment when error appears in the result of scanning invalid code. So you actually need to scan in order to reproduce.

**Pull request closed without merging**